base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -333,10 +324,10 @@ where D: Digest + Send + Sync
                                 self.curr_utxo_checkpoint.push_addition(k.clone());
                                 self.curr_range_proof_checkpoint.push_addition(proof_hash.clone());
                             }
-                            if let Some(index) = self.find_range_proof_leaf_index(proof_hash)? {
-                                lmdb_insert(&txn, &self.utxos_db, &k, &v)?;
-                                lmdb_insert(&txn, &self.txos_hash_to_index_db, &k, &index)?;
-                            }
+
+                            lmdb_insert(&txn, &self.utxos_db, &k, &v)?;
+                            let index = self.curr_range_proof_checkpoint.accumulated_nodes_added_count() - 1;
```

Comment:
This might have to use saturating_sub

Reply:
There will always be at least one node at this point in the code